runtime/onert/api/include/nnfw.h

```diff
@@ -257,6 +257,37 @@ NNFW_STATUS nnfw_prepare(nnfw_session *session);
  */
 NNFW_STATUS nnfw_run(nnfw_session *session);
 
+/**
+ * @brief     Run inference asynchronously
+ *
+ * <p>This function must be called after model is loaded by {@link nnfw_load_model_from_file},
+ * session is prepared for inference by {@link nnfw_prepare}, set input and output buffers
+ * by {@link nnfw_set_input} and {@link nnfw_set_output}.</p>
+ *
+ * <p>This function returns immediately after starting a thread to run the inference.
+ * To get the result of it {@nnfw_run_wait} must be called</p>
+ *
+ * @param[in] session The session to run inference
+ * @return    @c NNFW_STATUS_NO_ERROR if successful
+ */
+NNFW_STATUS nnfw_run_async(nnfw_session *session);
```

Comment:
@wateret Can one session have many inference threads?

Reply:
No. I chose not to support that for now.

Discussed from https://github.com/Samsung/ONE/issues/2541#issuecomment-648522313

> Let's support single asynchronous first, assuming that only one session is used asynchronously but sequentially.

Reply:
I see. Thanks.

Reply:
@hseok-oh I would like keep discussing on https://github.com/Samsung/ONE/issues/2541#issuecomment-657505760 PTAL 😄